pandas/_libs/parsers.pyx

```diff
@@ -1176,19 +1176,34 @@ cdef class TextReader:
             return result, na_count
 
         elif is_integer_dtype(dtype):
-            try:
-                result, na_count = _try_int64(self.parser, i, start,
-                                              end, na_filter, na_hashset)
-                if user_dtype and na_count is not None:
-                    if na_count > 0:
-                        raise ValueError(f"Integer column has NA values in column {i}")
-            except OverflowError:
+            do_try_uint64 = False
+            if user_dtype and dtype == 'uint64':
+                do_try_uint64 = True
+            else:
+                try:
```

Comment:
the pattern here is to _try_dtype then if that fails try another one, can you just do that here (rather than all of this if/then logic).
e.g. add a `_try_uint64` is needed

Reply:
Hm I agree with @jreback, this is hard to read

Reply:
I also agree and added a `_try_uint64` in commit https://github.com/pandas-dev/pandas/commit/927ddad322b59975255ac0c8f07ace0898441fd2 like @jreback suggested. Were you also referring to the former version with  `do_try_uint64` @phofl ? The latest version is <https://github.com/SandroCasagrande/pandas/blob/5896e017ca2960e0d535c8c0a0b9db978377bc91/pandas/_libs/parsers.pyx#L1182-L1198>. Sorry, if I did not signify correctly that I performed changes and the newest version should be reviewed again. Can you please have a look @jreback ?